pkg/controller/gitopsconfig/controller.go

```diff
@@ -432,29 +432,47 @@ func (r *Reconciler) manageDeletion(instance *gitopsv1alpha1.GitOpsConfig) (reco
 		return reconcile.Result{}, xerrors.Errorf("GitOpsConfig finalizer unable to list owned jobs for %q: %w", instance.Name, err)
 	}
 
-	// If exactly 1 job exists, but it's blocked because of bad image, we
-	// assume the GitOpsConfig never managed to successfully deploy, so we can
-	// just delete the job, remove the finalizer, and be done (#216). It may be
-	// either action=create or action=delete job.
-	// If a job is blocked because of bad image, it only has one active pod
-	if len(jobs) == 1 && jobs[0].Status.Succeeded == 0 && jobs[0].Status.Failed == 0 && jobs[0].Status.Active == 1 {
-		status, err := jobContainerStatus(context.TODO(), r.client, &jobs[0])
-		if err != nil {
-			log.Error(err, "GitOpsConfig finalizer unable to get job pod's status", "instance", instance.Name)
-			return reconcile.Result{}, xerrors.Errorf("GitOpsConfig finalizer unable to get job pod's status for %q: %w", instance.Name, err)
-		}
-		log.Info("GitOpsConfig finalizer found one job", "instance", instance.Name, "podStatus", status)
-		safeReasons := []string{"ErrImagePull", "ImagePullBackOff", "InvalidImageName"}
-		if status != nil && status.Waiting != nil && containsString(safeReasons, status.Waiting.Reason) {
-			err = r.client.Delete(context.TODO(), &jobs[0], client.PropagationPolicy(metav1.DeletePropagationBackground))
+	// If a job with owner GitOpsConfig exists, but it's blocked because of bad
+	// image, we assume the GitOpsConfig never managed to successfully deploy,
+	// so we can just delete the job, remove the finalizer, and be done (#216).
+	// It may be either action=create or action=delete job.
+	// There can be more than one stuck job - such situation occurs when there
+	// was a GitOpsConfig with a Periodic trigger which created a few jobs. If
+	// there is a job with GitOpsConfig owner, we remove the finalizer just
+	// after removing all stuck jobs - because such job would block a deletion
```

Comment:
What about a situation when there are some stuck jobs and some non-stuck jobs? or do we know somehow that this can't happen?

Reply:
There can be completed jobs from before, and stuck jobs from current configuration (because of wrong image). In such situation only stuck jobs are deleted, the rest will be garbage-collected.